core/libp2p/security/plaintext/plaintext.cpp

```diff
@@ -16,13 +18,54 @@ namespace libp2p::security {
   void Plaintext::secureInbound(
       std::shared_ptr<connection::RawConnection> inbound,
       SecConnCallbackFunc cb) {
-    cb(std::make_shared<connection::PlaintextConnection>(std::move(inbound)));
+    auto session =
+        std::make_shared<PlaintextSession>(this->marshaller_, inbound, cb);
+
+    // wait until the initiator sends us their public key
+    session->recvKey([this, cb, inbound, session](crypto::PublicKey remoteKey) {
```

Comment:
```suggestion
    session->recvKey([this, cb = std::move(cb), inbound = std::move(inbound), session](crypto::PublicKey remoteKey) {
```

Reply:
Also, is it safe to capture `this` without `shared_from_this()`?

Reply:
Adaptor should exist whole program lifetime